cosmetic refactors in conditional_trainer

Simontwice opened the pull request:
# Description
Moved functions (without changing any line):

    _train_step
    calculate_loss_and_maybe_optimize
    _decode_samples
    _save_weights

Moved and renamed:

    _restore_weights -> _load_model_weights

Minor changes in:

    _log_train_stats : deleted hacks
    _train_step : deleted measuring time
    _batches_eval_step -> _eval_variant (renamed function) : renamed variables

Deleted old hack functions.

# Checklist
   - [ ] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [ ] Are all variables set with logical defaults that are backward compatible?
   - [ ] I attached link to neptune if it was necessary
   - [ ] Have you successfully executed the linter and tests?
   - [ ] Are all functions concise and don't require splitting?
   - [ ] Is there documentation for arguments and complex logic?
   - [ ] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [ ] Are all functions placed in appropriate files?
   - [ ] Is the PR name meaningful and descriptive?
   - [ ] Is PR description provided, or unnecessary?
  
Other questions:
   - [ ] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [ ] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [ ] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

crewtool commented:
Copy-paste this comment to description of this PR:
---
Moved functions (without changing any line):
- `_train_step`
- `calculate_loss_and_maybe_optimize`
- `_decode_samples`
- `_save_weights`

Moved and renamed: 
- `_restore_weights` -> `_load_model_weights`

Minor changes in:
-  `_log_train_stats` : deleted hacks
- `_train_step` : deleted measuring time
`_batches_eval_step` -> `_eval_variant` (renamed function) : renamed variables

Deleted old hack functions.

crewtool reviewed (changes requested):
Change description.

Simontwice commented:
> ## Copy-paste this comment to description of this PR:
> 
> Moved functions (without changing any line):
> 
>     * `_train_step`
> 
>     * `calculate_loss_and_maybe_optimize`
> 
>     * `_decode_samples`
> 
>     * `_save_weights`
> 
> 
> Moved and renamed:
> 
>     * `_restore_weights` -> `_load_model_weights`
> 
> 
> Minor changes in:
> 
>     * `_log_train_stats` : deleted hacks
> 
>     * `_train_step` : deleted measuring time
>       `_batches_eval_step` -> `_eval_variant` (renamed function) : renamed variables
> 
> 
> Deleted old hack functions.

Thank you very much!!
